packages/medusa/src/api/admin/api-keys/[id]/route.ts

```diff
@@ -20,6 +21,13 @@ export const GET = async (
     req.remoteQueryConfig.fields
   )
 
+  if (!apiKey) {
+    throw new MedusaError(
```

Comment:
sugg: This will be the behavior for all route endpoints, might be good to build this into refetchEntity helper

Reply:
Yeah I agree, probably worth going through all endpoints and apply it in 1 go. Let me open a ticket for it.